pyvista/plotting/widgets.py

```diff
@@ -892,9 +998,9 @@ def _the_callback(widget, event_id):
             index = widget.WIDGET_INDEX
             if hasattr(callback, '__call__'):
                 if num > 1:
-                    try_callback(callback, point, index)
+                    try_callback(callback, point, index, widget)
```

Comment:
We should have a keyword argument in the `add_sphere_widget` method called `pass_widget` with a default to False, and only pass the widget if True

And then we may want to implement that for all the widgets